rfc/rfc-0000-allow-resources-to-report-status.md

```diff
@@ -0,0 +1,335 @@
+        operator: 'DoesNotExist'
+
+---
+apiVersion: carto.run/v1alpha1
+kind: ClusterTemplate
+metadata:
+  name: servicer
+spec:
+  resourceStatusRef:
+    name: service-rule
+
+  template:
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: $(workload.metadata.name)$
+    spec:
+      ...
+
+---
+apiVersion: carto.run/v1alpha1
+kind: Workload
+metadata:
+  ...
+spec:
+  ...
+status:
+  conditions:
+    ...
+  resources:
+    - name: service-creator
+      outputs:
+        ...
+      stampedRef:
+        ...
+      templateRef:
+        apiVersion: carto.run/v1alpha1
+        kind: ClusterTemplate
+        name: servicer
+      stampedStatus:
+        status: 'Healthy'
+        fields:
+          - key: status.loadBalancer
+            value: {}
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+A stamped resource can be in one of three states:
+1. 'Healthy'
+2. 'Unhealthy'
+3. 'Unknown'
+
+It is up to each template author to determine what "success" and "failure" mean for the given resource the template is stamping out. "Unknown" strictly represents that Cartographer has not be able to determine success or failure.
+
+A `ClusterResourceStatusRule` requires defining matchers for both the `Healthy` state and the `Unhealthy` state.
+The matchers in the `Healthy` state are ANDed together, whereas the matchers in the `Unhealthy` state are ORed. That is, a resource is considered to be `Healthy` only if _all of_ the matchers are fulfilled, but is considered to be `Unhealthy` if _any one of_ the matchers are fulfilled.
+
+The supported matchers are `matchConditions` and `matchFields`. Each is an array, and an author can define either or both of these matchers for a given state.
+
+For `matchConditions`, authors need only define the `type` and `status` of the condition for Cartographer to read. In the owner status, the full condition will be reflected (with the `reason`, `message`, etc.)
+For `matchFields`, the spec is based off of `matchFields` in blueprint resource options. Authors need to define the `key` and `operator`, and potentially the `values` (if the operator is `In` or `NotIn`). Authors can also optionally define a `messagePath`, which is a path in the resource where Cartographer can pull off a message to display in the owner status.
+
+Each template will now have a new field in the spec `resourceStatusRuleRef` where authors can specify the `ClusterResourceStatusRule` to use for that template. If no `ClusterResourceStatusRule` is defined, Cartographer will default to listing the resource as `Healthy` once it has been successfully applied to the cluster and any relevant outputs have been read off the resource.
+
+Templates will now also have a status with a `Ready` condition which will be `True` if the referenced `ClusterResourceStatusRule` exists (or if no rule is referenced), and `False` otherwise. Blueprints will now wait to report that they are `Ready` until their referenced templates are also `Ready`.
```

Comment:
We can avoid needing to reconcile all of the template types by inlining the status rules